internal/model/rwfolder.go

```diff
@@ -620,6 +620,8 @@ func (p *rwFolder) handleDir(file protocol.FileInfo) {
 		l.Debugf("need dir\n\t%v\n\t%v", file, curFile)
 	}
 
+	// Which filemode bits to preserve
+	const retainBits = os.ModeSetgid | os.ModeSetuid | os.ModeSticky
```

Comment:
I'd put this among the other consts at the top of the file for better visibility.